subsys/bluetooth/controller/hci/hci.c

```diff
@@ -2975,6 +3050,58 @@ void hci_evt_encode(struct radio_pdu_node_rx *node_rx, struct net_buf *buf)
 	}
 }
 
+u8_t hci_evt_concat(struct radio_pdu_node_rx **node_rx, struct net_buf *buf)
+{
+	u8_t node_cnt = 0;
+   struct radio_pdu_node_rx *node_rx_slice[HCI_MAX_NR_OF_CONCAT_MSG] = {NULL};
+
+	while(node_rx[node_cnt] && (node_cnt < HCI_MAX_NR_OF_CONCAT_MSG))
+	{
+		switch (node_rx[node_cnt]->hdr.type)
+		{
+			case NODE_RX_TYPE_REPORT:
+         {
+            if(is_direct_adv_report(node_rx[node_cnt])) {
+               if(node_cnt > 0) {
+                 /* Transmit previous packets - single direct packet
+                  * will be processed during next hci_evt_concat() call
+                  */
+                  le_advertising_report(node_rx_slice, buf);
+                  return node_cnt;
+               }
+
+               /* Process only single direct packet */
+               le_direct_advertising_report(node_rx[node_cnt], buf);
+               return 1;
+            }
+            break;
+         }
+
+			default:
+         {
+            if(node_cnt > 0) {
+               /* Transmit previous packets - single non type report packet
+                * will be processed during next hci_evt_concat() call
+                */
+               le_advertising_report(node_rx_slice, buf);
+               return node_cnt;
+            }
+
+            /* Process only single non advertising packet */
+            encode_control(node_rx[node_cnt], (void *) node_rx[node_cnt]->pdu_data, buf);
+            return 1;
+            break;
+         }
+		}
+
+		node_rx_slice[node_cnt] = node_rx[node_cnt];
+		node_cnt++;
+	}
+
+	le_advertising_report(node_rx, buf);
```

Comment:
I'd rather change the argument to `node_rx_slice`, and `goto` here from lines 3065 & 3082, avoiding copy pasta.